src/elements/frame_element.js

```diff
@@ -82,6 +82,10 @@ export class FrameElement extends HTMLElement {
     return this.getAttribute("refresh")
   }
 
+  get shouldReloadWithMorph() {
+    this.src && this.refresh === "morph"
```

Comment:
Should this have a `return` statement?

Reply:
Also, is it important to exclude initial `<turbo-frame>` elements that don't have a `[src]` attribute? Wouldn't morphing an initial HTTP request have a benefit?

Reply:
You're too fast for me! :) I actually mistakenly pushed before I was done. I'm doing a little more cleanup and making sure all tests pass before I went ahead and tagged anyone.

However, I don't understand your second comment. My mental model was that morphing for turbo frames only applied when there was a src. And I didn't introduce that extra condition, that's currently present within main.

When there is no src, then the full contents of the HTML must be supplied within the tags so that means it's part of the overall page. And so if the overall page morphs, then there is no special handling for turbo-frames — the contents within them just morphs along with all the rest of the HTML of that page. But I could be misunderstanding that.

That was my read of this original logic:
https://github.com/hotwired/turbo/blob/main/src/core/drive/morph_renderer.js#L95

The event binding hack was only being added when there was a src so I assume without that it's just regular HTML morphing of the page.